tests/codebuild/run_canarytest.sh

```diff
@@ -49,11 +49,20 @@ echo "Launching canary test for ${COMMIT_SHA}"
 
 # Launch EKS cluster if we need to and define cluster_name,CLUSTER_REGION.
 echo "Launching the cluster"
-readonly cluster_name="sagemaker-k8s-pipeline-"$(date '+%Y-%m-%d-%H-%M-%S')""
 
-# By default eksctl picks random AZ, which time to time leads to  capacity issue.
-# Generally 1a, 1b, 1c are topmost available AZ, hence specifying it explicitly 
-eksctl create cluster "${cluster_name}" --nodes 1 --node-type=c5.xlarge --timeout=40m --region "${CLUSTER_REGION}" --auto-kubeconfig --version ${CLUSTER_VERSION} 
+cluster_name="sagemaker-k8s-canary-"$(date '+%Y-%m-%d-%H-%M-%S')""
+
+if [ -z "${NON_EPHEMERAL_CLUSTER}" ]
```

Comment:
Could we make this a more generic variable name?
Something like `USE_EXISTING_CLUSTER`?

Reply:
I think this name is good and explanatory enough.

Reply:
++ to Nick's suggestion; it more accurately describes the purpose as this code uses any existing cluster, not necessarily a non-ephemeral one.